pkg/store/bucket.go

```diff
@@ -132,6 +143,15 @@ func newBucketStoreMetrics(reg prometheus.Registerer) *bucketStoreMetrics {
 		},
 	})
 
+	m.queriesDropped = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "thanos_bucket_store_queries_dropped_total",
+		Help: "Number of queries that were dropped due to the sample limit.",
+	})
+	m.queriesLimit = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "thanos_bucket_store_queries_limit",
```

Comment:
Prometheus' team use `prometheus_engine_queries_concurrent_max`. Naming is hard but I figure we could be consistent with them ?

https://github.com/prometheus/prometheus/blob/8155cc49924cd48ec552506b4e5519ebafa8b722/promql/engine.go#L234-L239

Reply:
Good point. Should rename this to `thanos_bucket_store_queries_concurrent_max`.